commit 23c10d8816d4dfb4feca6347150b5c56e0ae5eda
Author: Ambroz Bizjak <ambrop7@gmail.com>
Date:   Thu Jan 5 19:06:51 2017 +0100

    Renaming min and max macros to fix breakage of C++ algorithm.

diff --git a/sam/drivers/uotghs/uotghs_device.h b/sam/drivers/uotghs/uotghs_device.h
index 560d1c8..3f8f130 100644
--- a/sam/drivers/uotghs/uotghs_device.h
+++ b/sam/drivers/uotghs/uotghs_device.h
@@ -291,7 +291,7 @@ extern "C" {
   //! Bounds given integer size to allowed range and rounds it up to the nearest
   //! available greater size, then applies register format of UOTGHS controller
   //! for endpoint size bit-field.
-#define udd_format_endpoint_size(size)            (32 - clz(((uint32_t)min(max(size, 8), 1024) << 1) - 1) - 1 - 3)
+#define udd_format_endpoint_size(size)            (32 - clz(((uint32_t)atmel_min(atmel_max(size, 8), 1024) << 1) - 1) - 1 - 3)
   //! Configures the selected endpoint size
 #define udd_configure_endpoint_size(ep, size)     (Wr_bitfield(UOTGHS_ARRAY(UOTGHS_DEVEPTCFG[0], ep), UOTGHS_DEVEPTCFG_EPSIZE_Msk, udd_format_endpoint_size(size)))
   //! Gets the configured selected endpoint size
diff --git a/sam/utils/compiler.h b/sam/utils/compiler.h
index 201fa7e..f3eec4b 100644
--- a/sam/utils/compiler.h
+++ b/sam/utils/compiler.h
@@ -817,7 +817,7 @@ typedef struct
  *
  * \note More optimized if only used with values unknown at compile time.
  */
-#define min(a, b)   Min(a, b)
+#define atmel_min(a, b)   Min(a, b)
 
 /*! \brief Takes the maximal value of \a a and \a b.
  *
@@ -828,7 +828,7 @@ typedef struct
  *
  * \note More optimized if only used with values unknown at compile time.
  */
-#define max(a, b)   Max(a, b)
+#define atmel_max(a, b)   Max(a, b)
 
 //! @}
 
